core/env/clone_spec.rb

```diff
@@ -0,0 +1,29 @@
+require_relative '../../spec_helper'
+
+describe "ENV#clone" do
+  ruby_version_is "3.2" do
+    it "raises TypError" do
+      message = /Cannot clone ENV, use ENV.to_h to get a copy of ENV as a hash/
+
+      -> {
+        ENV.clone(freeze: nil)
+      }.should raise_error(TypeError, message)
```

Comment:
**minor:** Not sure whether this check (with `freeze: nil`) is needed. Could you elaboration on it?